lua/entities/gmod_tardis/modules/sh_tracking.lua

```diff
@@ -0,0 +1,497 @@
+            end
+        end
+
+        if IsValid(ply) then
+            if wasTracking ~= isTracking then
+                TARDIS:StatusMessage(ply, "Controls.Tracking.Status", IsValid(ent))
+            end
+            if isTracking then
+                if wasTrackingEnt ~= ent then
+                    local name = ent.PrintName or (isfunction(ent.Name) and ent:Name()) or ent.Name or ent:GetModel() or ent:GetClass()
+                    if ent.GetCreator then
+                        local creator = ent:GetCreator()
+                        if IsValid(creator) then
+                            name = name .. " (" .. creator:Nick() .. ")"
+                        end
+                    end
+                    TARDIS:Message(ply, "%s %s", "Controls.Tracking.Target", name)
+                else 
+                    TARDIS:Message(ply, "Controls.Tracking.SameTarget")
+                end
+            end
+        end
+        return true
+    end
+
+    function ENT:GetTrackRotation()
+        return self:GetData("tracking-rotation", false)
+    end
+
+    function ENT:SetTrackRotation(on)
+        self:SetData("tracking-rotation", on)
+
+        local ent = self:GetTracking()
+        if not IsValid(ent) then return end
+        local offsetPos = self:GetData("tracking-offset-pos", Vector(0,0,0))
+        local offsetYaw = self:GetData("tracking-offset-yaw", 0)
+        local newOffsetPos, newOffsetYaw
+        if on then
+            local currentTrackingPos = ent:GetPos() + offsetPos
+            newOffsetPos = ent:WorldToLocal(currentTrackingPos)
+            newOffsetYaw = offsetYaw - ent:GetAngles().y
+        else
+            local currentTrackingPos = ent:LocalToWorld(offsetPos)
+            newOffsetPos = WorldToLocal(currentTrackingPos, angle_zero, ent:GetPos(), angle_zero)
+            newOffsetYaw = offsetYaw + ent:GetAngles().y
+        end
+        self:SetData("tracking-offset-pos", newOffsetPos)
+        self:SetData("tracking-offset-yaw", newOffsetYaw)
+    end
+
+    function ENT:ToggleTrackRotation()
+        self:SetTrackRotation(not self:GetTrackRotation())
+    end
+
+    ENT:AddHook("Initialize", "tracking", function(self)
+        self:SetTrackRotation(true)
+    end)
+
+    ENT:AddHook("HandleE2", "tracking", function(self, name, e2, ...)
+        local args = {...}
+        if name == "Track" and (args[1] == e2.player or TARDIS:CheckPP(e2.player, self)) then
+            local ent = args[1]
+            local success = self:SetTracking(ent, e2.player)
+            return success and 1 or 0
+        end
+    end)
+
+    ENT:AddHook("HandleE2", "tracking_get", function(self, name, e2)
+        if name == "GetTracking" then
+            return self:GetTracking() or NULL
+        end
+    end)
+
+    ENT:AddHook("FlightToggled", "tracking", function(self, on)
+        if not on then
+            self:SetTracking(nil, self:GetData("pilot"))
+        end
+    end)
+
+    local VECTOR_0_0_1 = Vector(0,0,1)
```

Comment:
Unimportant nit-pick: could probably just call it VECTOR_UP, would make the code easier to read

Reply:
https://github.com/MattJeanes/TARDIS/pull/969/commits/c0961ad6060f3421d2d38c467ed3b5ebf776a429